### PR TITLE
[12.0][FIX] barcodes_generator_[abstract,product]: Respect create vals

### LINKED ISSUE #538: barcodes_generator_abstract: 'barcode' value not respected in create() of 'barcode.generate.mixin'
## Module
barcodes_generator_abstract (in combination with barcodes_generator_product).

## Describe the bug
When creating a product and manually typing in the barcode, that barcode is not saved to the product if barcode automation is active.

## To Reproduce
**Affected versions**: 12, probably later.

1. Install barcodes_generator_product
2. Enable automation (with sequence) on the product barcode.rule.
3. Install `pos_price_to_weight`, which has a demo product with a defined barcode.
4. Error during installation.

**Expected behavior**
If a barcode is defined during creation, we probably shouldn't override it with automation.

**Additional context**
I'm working on seeing if the module is compatible with `pos_price_to_weight`.

### LINKED ISSUE #539: barcodes_generator_abstract: 'barcode_rule_id' value not respected in create() of 'barcode.generate.mixin'
## Module
barcodes_generator_abstract (in combination with barcodes_generator_product).

## Describe the bug
When creating a product and manually selecting the barcode rule during creation, that rule is not applied if another barcode rule is set as automated.

## To Reproduce
**Affected versions**: 12, probably later.

1. Install barcodes_generator_product
2. Enable automation (with sequence) on the product module's barcode.rule.
3. Create a new barcode.rule for products. Enable generation for it (optional, but makes testing easier).
4. Create a new product. During product creation, select the newly created barcode.rule.
5. The module's barcode.rule generation is applied, but the new barcode.rule is saved to the product. If you click on 'generate barcode', the new barcode.rule's generation is applied.

**Expected behavior**
The automated barcode.rule generation is not applied to products that specify a different barcode.rule generation. Optionally, if the specified barcode.rule allows generation, apply it automatically. Not optionally, if the specified barcode.rule _is_ the one with automation, then apply the generation.

**Additional context**
I'm working on seeing if the module is compatible with `pos_price_to_weight`.

Related to #538

### PR DESCRIPTION
**Pull request by carmenbianca**
Fixes #538 
Fixes #539 

Tagging @legalsylvain

### CODE REVIEW

File: barcodes_generator_abstract/models/barcode_generate_mixin.py [hunk 1]
```diff
@@ -36,13 +36,16 @@ class BarcodeGenerateMixin(models.AbstractModel):
...
-                'barcode_rule_id': barcode_rule.id,
-            })
+        automatic = False
+        if not vals.get("barcode") and not vals.get("barcode_rule_id"):
+            barcode_rule = self.env['barcode.rule'].get_automatic_rule(self._name)
+            if barcode_rule.exists():
+                automatic = True
+                vals.update({
+                    'barcode_rule_id': barcode_rule.id,
+                })
```
Comment (legalsylvain) [inline barcodes_generator_abstract/models/barcode_generate_mixin.py]:
```suggestion
        barcode_rule = False
        if not vals.get("barcode") and not vals.get("barcode_rule_id"):
            barcode_rule = self.env['barcode.rule'].get_automatic_rule(self._name)
            if barcode_rule.exists():
                vals.update({
                    'barcode_rule_id': barcode_rule.id,
                })
```

File: barcodes_generator_abstract/models/barcode_generate_mixin.py [hunk 2]
```diff
@@ -36,13 +36,16 @@ class BarcodeGenerateMixin(models.AbstractModel):
...
         record = super(BarcodeGenerateMixin, self).create(vals)
-        if barcode_rule:
+        if automatic:
```
Comment (legalsylvain) [inline barcodes_generator_abstract/models/barcode_generate_mixin.py]:
```suggestion
        if barcode_rule.exists():
```

Comment (carmenbianca) [reply barcodes_generator_abstract/models/barcode_generate_mixin.py]: `False.exists()` does not exist. Is `exists()` pertinent here?

**Review by legalsylvain — APPROVED** [review]

File: barcodes_generator_abstract/models/barcode_generate_mixin.py (hunk 2 above)
Comment (legalsylvain) [reply barcodes_generator_abstract/models/barcode_generate_mixin.py]: ow ! you're right. 
Maybe  barcode_rule = self.env['barcode.rule'] instead of False.

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@carmenbianca Is this ready ?

**Comment by carmenbianca** [comment]
@rousseldenis yes, it's ready :D

**Pull request merged**